pennylane/measurements/probs.py

```diff
@@ -259,23 +259,21 @@ def _count_samples(indices, batch_size, dim):
         """Count the occurrences of sampled indices and convert them to relative
         counts in order to estimate their occurrence probability."""
         num_bins, bin_size = indices.shape[-2:]
-        if batch_size is None:
-            prob = qml.math.zeros((dim, num_bins), dtype="float64")
-            # count the basis state occurrences, and construct the probability vector for each bin
-            for b, idx in enumerate(indices):
-                basis_states, counts = qml.math.unique(idx, return_counts=True)
-                prob[basis_states, b] = counts / bin_size
+        interface = qml.math.get_deep_interface(indices)
 
-            return prob
+        def _count_samples_core(indices, dim, interface):
+            return qml.math.array(
+                [[qml.math.sum(idx == p) for p in range(dim)] for idx in indices], like=interface
```

Comment:
If we `swapaxes` on any usage of `_count_samples_core` anyways, couldn't we do this and skip the swapping?
```suggestion
                [[qml.math.sum(idx == p) for idx in indices] for p in range(dim)], like=interface
```

Reply:
I think I went with that to loop over the contiguous index, but it's probably pointless for a Python list.

Reply:
Ah, good thinking! I would _think_ that for lists it indeed doesn't matter :thinking: